File: Week_1/Day_4/Reshape.py
class Solution:
    def matrixReshape(self, matrix: list[list[int]], rows: int, cols: int) -> list[list[int]]:
        m, n = len(matrix), len(matrix[0])

        if m * n != rows * cols:
            return matrix

        result = [[0] * cols for _ in range(rows)]
        index = 0

        for i in range(m):
            for j in range(n):
                result[index // cols][index % cols] = matrix[i][j]
                index += 1

        return result

File: Week_1/Day_4/Reshape.java
class Solution {
    public int[][] matrixReshape(int[][] matrix, int rows, int cols) {
        int m = matrix.length;
        int n = matrix[0].length;

        if (m * n != rows * cols) return matrix;

        int[][] result = new int[rows][cols];
        int index = 0;

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                result[index / cols][index % cols] = matrix[i][j];
                index++;
            }
        }

        return result;
    }
}